visualization/app/codeCharta/util/nodeDecorator.spec.ts

```diff
@@ -44,15 +36,16 @@ describe("nodeDecorator", () => {
 		NodeDecorator.decorateMapWithPathAttribute(file)
 	})
 
-	function allUniqueIds(map: HierarchyNode<CodeMapNode>): boolean {
+	function allUniqueIds(map: HierarchyNode<CodeMapNode>) {
 		const ids = new Set()
+		let isUnique = true
 		map.each(node => {
 			if (ids.has(node.id)) {
-				return false
+				isUnique = false
 			}
 			ids.add(node.id)
 		})
-		return true
+		return isUnique
```

Comment:
Fixed uniqueness check

Reply:
Why is this fixed? We check if ALL nodes have unique ids here. We can return as soon as we encounter an id, that we've seen already. Your suggestion is working the same, but it's traversing through all nodes instead of returning earlier.

Reply:
As a hint: it's not a for loop and there's a function scope :-)